src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const changesArray = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    }
+
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
```

Comment:
The `addProperties` action is correctly handled by merging `extraData` into the `currentState`. Ensure that `extraData` is always an object to avoid runtime errors.